internal/controller/ai_gateway_route.go

```diff
@@ -94,11 +126,34 @@ func (c *AIGatewayRouteController) Reconcile(ctx context.Context, req reconcile.
 	// TODO: merge this into syncAIGatewayRoute. This is a left over from the previous sink based implementation.
 	c.logger.Info("Ensuring extproc configmap exists", "namespace", aiGatewayRoute.Namespace, "name", aiGatewayRoute.Name)
 	if err := c.ensuresExtProcConfigMapExists(ctx, &aiGatewayRoute); err != nil {
+		c.logger.Error(err, "failed to reconcile extProc config map")
+		condition := metav1.Condition{
+			Type:    aiGatewayRouteConditionTypeReconciled,
+			Status:  metav1.ConditionFalse,
+			Reason:  aiGatewayRouteReasonConfigMapError,
+			Message: fmt.Sprintf("failed to reconcile config map: %v", err),
+		}
+		if err = c.patchAIGatewayRouteStatus(ctx, &aiGatewayRoute, condition); err != nil {
+			c.logger.Error(err, "failed to patch AIGatewayRoute status")
+		}
```

Comment:
can you extract this as a method which accepts necessary stuff here

Reply:
OK, I put those into patchAIGatewayRouteStatus().